forte/data/data_store.py

```diff
@@ -313,52 +399,82 @@ def delete_entry(self, tid: int):
             tid (int): Unique id of the entry.
         """
         # We retrieve the entry data from `__entry_dict` using tid. We get its
-        # `entry_type`, `type_id`, `begin` and `end` indices. we remove the
-        # entry from `__entry_dict` when its information is retrieved. We
-        # find the
-        # `entry_type` sortedlist using `type_id`. We bisect the sortedlist to
-        # find the index of entry data. Then, `type_id` and index are passed to
-        # _delete_entry_by_loc.
+        # `entry_type`, `type_id`. If it is an Annotation-Like entry, we obtain
+        # `begin` and `end` indices and use bisection to locate the `index_id`.
+        # If it is not, we obtain the 'index_id' directly from `__entry_dict`.
+        # We remove the entry from `__entry_dict` when its information is
+        # retrieved. Then, the `type_id` and the `index_id` are passed to
```

Comment:
NO, given a tid we should always find it directly. Obtaining the `begin` and `end` through a list and do bisect is very indirect. Think about it, If you have a mapping to `begin` and `end` why can't we have a mapping to its position?

Reply:
You already have a function `get_entry` below, just call it from here.

Reply:
yes